src/api/services/apiKeyService.ts

```diff
@@ -0,0 +1,41 @@
+import { Participant } from '../entities/Participant';
+import { CreatedApiKeyDTO } from './adminServiceHelpers';
+
+export type ApiKeySecretsDTO = {
+  plaintextKey: string;
+  secret: string;
+  name: string;
+};
+
+export const createdApiKeyToApiKeySecrets = (apiKey: CreatedApiKeyDTO): ApiKeySecretsDTO => {
+  // eslint-disable-next-line camelcase
+  return {
+    plaintextKey: apiKey.plaintext_key,
+    secret: apiKey.authorizable.secret,
+    name: apiKey.authorizable.name,
+  };
+};
+
+export const getApiRoles = async (participant: Participant) => {
+  const participantCurr = await Participant.query()
+    .findById(participant.id)
+    .withGraphFetched('apiRoles');
+
+  if (!participantCurr?.apiRoles) {
+    return [];
+  }
+  return participantCurr.apiRoles;
+};
+
+export const checkApiRoles = async (
+  keyRoles: string[],
+  participant: Participant
+): Promise<boolean> => {
+  const userRoles = (await getApiRoles(participant)).map((role) => role.roleName);
```

Comment:
maybe `participantRoles` is more clear here - we may introduce user roles in the future so we want to avoid possible confusion.